api/src/feeds/impl/feeds_api_impl.py

```diff
@@ -155,6 +161,21 @@ def get_gtfs_feeds(
             ),
         )
         gtfs_feed_query = gtfs_feed_filter.filter(Database().get_query_model(Gtfsfeed))
+
+        # Create a subquery for the latest datasets to avoid loading all datasets
+        latest_datasets_subquery = select(Gtfsdataset).filter(Gtfsdataset.latest).subquery()
+        latest_datasets = aliased(Gtfsdataset, latest_datasets_subquery)
+
+        gtfs_feed_query = (
+            gtfs_feed_query.outerjoin(Location, Feed.locations)
+            .outerjoin(latest_datasets, Gtfsfeed.gtfsdatasets)
+            .options(
+                contains_eager(Gtfsfeed.gtfsdatasets, alias=latest_datasets)
+                .joinedload(Gtfsdataset.validation_reports)
+                .joinedload(Validationreport.notices),
+                *BasicFeedImpl.get_joinedload_options(),
+            )
+        )
```

Comment:
How can we benefit from the latest datasets query if the _GtfsRTFeedImpl.from_orm_ is still looping over all datasets?

Reply:
Indeed! There's no way to load the latest dataset exclusively so I removed this query.

Reply:
We will hit a performance issue in the future as the datasets keep growing. We can revisit mapping the latest dataset as part of the entity. I remember having a few challenges with the previous SqlAchemy version; maybe the new version is easier to implement. Created an issue for it #548